.github/workflows/app.yml

```
@@ -0,0 +1,30 @@
+name: Test App
+
+on:
+  push:
+    branches:
+      - master
+      - release-*
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Setup Gradle cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+      - name: Run test and assemble tasks
+        run: ./gradlew --no-daemon --build-cache test assemble
```

```suggestion
        run: ./gradlew --no-daemon --build-cache test
```
AFAIK assemble should be in the publish task and doesn't provide much extra on top of test except for packaging the .apk file. The test job will compile all code already.

---

It does do minification and packaging steps, which might show errors in specific cases.. but I guess just using test should be enough.